mw_redis_cache.go

```diff
@@ -37,15 +37,7 @@ func (m *RedisCacheMiddleware) Init() {
 }
 
 func (m *RedisCacheMiddleware) EnabledForSpec() bool {
-	if !m.Spec.CacheOptions.EnableCache {
-		return false
-	}
-	for _, version := range m.Spec.VersionData.Versions {
```

Comment:
I think it was sort of optimization here, to not run this middleware all the time. 
And it could be fixed without removing this `for` loop and adding an additional check like:
```
if m.Spec.CacheOptions.CacheAllSafeRequests {
   return true
}
```

Reply:
Sure - but with this PR, the middleware does *not* run all the time. It only runs if caching is enabled.

And when it's enabled but only for some cases, the cases that end up not having the caching run the middleware for very little time. They are discarded very early on in ProcessRequest.

I think it is best if we just keep it simple, as we have clearly fucked it up once already. And the even more complex logic would still have edge cases. For example, what if caching is enabled, but `CacheAllSafeRequests` is false and there are no cached extended paths? I'm not that sure that the middleware should be removed entirely in such a case.

Or, to put it another way, what would `EnableCache` even be for, if we didn't rely on it here?

Reply:
Yeah, I'm ok with the current solution too, it should work well. Just needs to be confirmed.

Reply:
To clarify - I also started tweaking the current heuristic, instead of just reducing it to the one bool field. But after thinking about it a little bit, I realised it was silly and overly complex.